src/vmshepherd/http/rpc_api.py

```diff
@@ -112,13 +114,23 @@ def wrap(self, *args, **kwargs):
         :rtype: string
         """
         vmshepherd = self.request.app.vmshepherd
-        preset = vmshepherd.preset_manager.get_preset(preset_name)
+        try:
+            preset = vmshepherd.preset_manager.get_preset(preset_name)
+        except PresetNotFound as ex:
+            logging.exception(ex)
+            raise
+
         # check in cache
         for vm in preset.vms:
             if vm_id == vm.id:
                 return {'vm_ip': vm.ip[0]}
         # retrieve real time data
-        vm_info = await preset.iaas.get_vm(vm_id)
+        try:
+           vm_info = await preset.iaas.get_vm(vm_id)
+        except VmNotFound as ex:
+            logging.exception(ex)
```

Comment:
Does VMNotFound really needs a stacktrace?

Reply:
It does not, solved